src/test/java/nugget/TaskListTest.java

```diff
@@ -0,0 +1,77 @@
+package nugget;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class TaskListTest {
+
+    private TaskList taskList;
+    private Task todoTask;
+    private Task deadlineTask;
+
+    @BeforeEach
+    void setUp() {
+        taskList = new TaskList();
+        todoTask = new Todo("Read book");
+        deadlineTask = new Deadline("Submit assignment", "2024-09-01 0900");
+    }
+
+    @Test
+    void addTask_shouldAddTaskToTaskList() {
+        taskList.addTask(todoTask);
+        assertEquals(1, taskList.size(), "Task list size should be 1 after adding a task");
+        assertEquals(todoTask, taskList.getTask(0), "Added task should be the same as the one retrieved");
+    }
+
+    @Test
+    void deleteTask_shouldRemoveTaskFromTaskList() {
+        taskList.addTask(todoTask);
+        taskList.addTask(deadlineTask);
+        taskList.deleteTask(0);
+        assertEquals(1, taskList.size(), "Task list size should be 1 after deleting a task");
+        assertEquals(deadlineTask, taskList.getTask(0), "Remaining task should be the deadline task after deleting the first task");
+    }
+
+    @Test
+    void markTask_shouldMarkTaskAsCompleted() {
+        taskList.addTask(todoTask);
+        taskList.markTask(0);
+        assertTrue(taskList.getTask(0).isCompleted(), "Task should be marked as completed");
+    }
+
+    @Test
```

Comment:
For the test methods, maybe a better name would be in the form of unitBeingTested_descriptionOfTestInputs_expectedOutcome (as specified from the notes)?

Reply:
I agree with this comment.

Reply:
Thank you for pointing that out!